crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -143,8 +143,17 @@ impl BaseClient {
                 room_info.mark_as_joined(); // FIXME: this might not be accurate
                 room_info.mark_state_partially_synced();
 
-                // FIXME not yet supported by sliding sync.
-                // room_info.update_summary(&room_data.summary);
+                // RoomSummart not yet supported by sliding sync.
+                // so I am creating a fake one and setting the invited
+                // and joined member values
+                let mut room_summary = RoomSummary::new();
+                let joined = store.get_joined_user_ids(room_id).await?;
+                let invited = store.get_invited_user_ids(room_id).await?;
+
+                room_summary.invited_member_count = UInt::new(invited.len() as u64);
+                room_summary.joined_member_count = UInt::new(joined.len() as u64);
+
+                room_info.update_summary(&room_summary);
```

Comment:
Usually after a while that the application is running it gives me always the right value, so yeah it's probably filling the values alongside the SS.
This means that value is not always accurate and depends on the state of the SS.

Reply:
Ah sorry I might have been a bit unclear. This is not what I meant, you're fetching the counts from storage now, and this works once you fetched all the `m.room.member` events.

What I meant is that we should fetch the counts from the sliding sync response, the counts are in a different place in the new response type and have not been connected to be fetched and put in the right place in the `RoomInfo` struct.

Here's a diff that explains things a bit better:

```diff
diff --git a/crates/matrix-sdk-base/src/sliding_sync.rs b/crates/matrix-sdk-base/src/sliding_sync.rs
index 5f74bd2118..5eefadd680 100644
--- a/crates/matrix-sdk-base/src/sliding_sync.rs
+++ b/crates/matrix-sdk-base/src/sliding_sync.rs
@@ -143,15 +143,15 @@ impl BaseClient {
                 room_info.mark_as_joined(); // FIXME: this might not be accurate
                 room_info.mark_state_partially_synced();
 
-                // RoomSummart not yet supported by sliding sync.
-                // so I am creating a fake one and setting the invited
-                // and joined member values
+                // Sliding sync doesn't have a room summary, nevertheless it contains the joined
+                // and invited member counts. It likely will never have a heroes concept since it
+                // calculates the room display name for us.
+                //
+                // Let's at least fetch the member counts, since they might be useful.
                 let mut room_summary = RoomSummary::new();
-                let joined = store.get_joined_user_ids(room_id).await?;
-                let invited = store.get_invited_user_ids(room_id).await?;
 
-                room_summary.invited_member_count = UInt::new(invited.len() as u64);
-                room_summary.joined_member_count = UInt::new(joined.len() as u64);
+                room_summary.invited_member_count = room_data.invited_count;
+                room_summary.joined_member_count = room_data.joined_count;
 
                 room_info.update_summary(&room_summary);
 
```

Reply:
Oh okay I thought what you meant  the store data, but actually we wanted the one in the response directly.
Thanks a lot!

I have tested this... and it kinda... works?
The number stays the same, it doesn't change now but it's always more than the expected number

For some weird reason the function returns more members than there are, for example i tested this in a room where my account, bigtestaccount and Doug are present, and it returned 4, instead of 3.
Same for other rooms for example Matrix HQ returns: 44949 but in reality it has 44873.

Could it be that there is something wrong in the `room_data.invited_count/joined_count` values? Or in how the `active_members_count` is calculated? Tomorrow I'll look a bit more into this.